lntest/harness_assertion.go

```diff
@@ -377,6 +377,22 @@ func (h *HarnessTest) AssertChannelExists(hn *node.HarnessNode,
 	return channel
 }
 
+// AssertOutputScriptClass checks that the specified transaction output has the
+// expected script class.
+func (h *HarnessTest) AssertOutputScriptClass(tx *btcutil.Tx,
+	outputIndex uint32, scriptClass txscript.ScriptClass) {
+
+	require.Greater(h, len(tx.MsgTx().TxOut), int(outputIndex))
+
+	txOut := tx.MsgTx().TxOut[outputIndex]
+
+	class, _, _, err := txscript.ExtractPkScriptAddrs(
```

Comment:
Can use `txscript.ParsePkScript` instead which returns a `PkScript` that has a `Class()` method.